string_unicode_test.go

```diff
@@ -0,0 +1,41 @@
+package goja
+
+import (
+	"testing"
+)
+
+func TestStringUnicodeMemUsage(t *testing.T) {
+	tests := []struct {
+		name        string
+		val         unicodeString
+		expected    uint64
+		newExpected uint64
+		errExpected error
+	}{
+		{
+			name:        "should have a value given the length of the string",
+			val:         unicodeString{' ', 'y', 'o'},
+			expected:    2,
```

Comment:
[q] why 2 instead of 3? Does space not count?

Reply:
When this gets converted to a string (see the String() method on this type) we skip the first item which can represent the little-indian/big-indian convention but not the actual content of the string. Not 100% that's what the first char is used for but I know that even comparisons between strings skip the fist char if it's a unicode string.